File: sloth/src/main/kotlin/com/xiansenliu/sloth/OnDeniedCallback.java
package com.xiansenliu.sloth;

import java.util.List;

/**
 * Author       xinliu
 * Date         6/24/17
 * Time         3:17 PM
 */

public interface OnDeniedCallback {

    void onDenied(int requestCode, List<String> deniedPermissions, Action goSettingAction);
}

File: sloth/src/main/kotlin/com/xiansenliu/sloth/OnDeniedCallback.kt
package com.xiansenliu.sloth

/**
 * Author       xinliu
 * Date         6/24/17
 * Time         3:17 PM
 */

interface OnDeniedCallback {

    fun onDenied(requestCode: Int, deniedPermissions: List<String>, goSettingAction: Action)

}

internal val DEFAULT_ON_DENIED = object : OnDeniedCallback {
    override fun onDenied(requestCode: Int, deniedPermissions: List<String>, goSettingAction: Action) {
        goSettingAction.invoke()
    }
}

